core/src/main/scala/org/apache/spark/storage/RDDInfo.scala

```diff
@@ -49,9 +50,40 @@ class RDDInfo(
   }
 }
 
+
 private[spark] object RDDInfo {
   def fromRdd(rdd: RDD[_]): RDDInfo = {
     val rddName = Option(rdd.name).getOrElse(rdd.id.toString)
     new RDDInfo(rdd.id, rddName, rdd.partitions.size, rdd.getStorageLevel)
   }
 }
+
+@DeveloperApi
+class BroadcastInfo(
+    val id: Long,
+    val name: String,
+    val numPartitions: Int) extends Ordered[BroadcastInfo] with InMemoryObjectInfo {
+
+  var memSize = 0L
+  var diskSize = 0L
+  var tachyonSize = 0L
+
+  override def toString = {
+    import Utils.bytesToString
+    ("%s\" (%d) ; " +
+      "MemorySize: %s; TachyonSize: %s; DiskSize: %s").format(
+        name, id, bytesToString(memSize), bytesToString(tachyonSize), bytesToString(diskSize))
+  }
+
+  override def compare(that: BroadcastInfo): Int = {
+    if (this.id > that.id) {
+      1
+    } else {
+      if (this.id == that.id) {
+        return 0
+      }
+      -1
```

Comment:
If you really want a suggestion here :-):

    com.google.common.primitives.Longs.compare(this.id, that.id)